Build fails with Xcode 15.3 beta

I see C++ errors with both Xcode 15.3b1 and b2.

```
$ DEVELOPER_DIR=/Applications/Xcode15.3b1.app make macos
set -o pipefail && /usr/bin/xcodebuild build -project "MoltenVKPackaging.xcodeproj" -scheme "MoltenVK Package (macOS only)" -destination "generic/platform=macOS" GCC_PREPROCESSOR_DEFINITIONS='${inherited} ' -quiet
2024-02-08 22:25:38.395 xcodebuild[86472:54001714] Requested but did not find extension point with identifier Xcode.DVTDeviceFoundation.Aspect for extension Xcode.IDEiOSSupportCore.Aspect.NBDD of plug-in com.apple.dt.IDE.IDEiOSSupportCore
2024-02-08 22:25:38.397 xcodebuild[86472:54001714] Requested but did not find extension point with identifier DVTProducts.ReleaseChannelNamer for extension DVTProducts.ReleaseChannelNamer.PV of plug-in com.apple.dt.IDEDistributionSupport
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.m:33:17: warning: property 'lodBiasMVK' requires method 'lodBiasMVK' to be defined - use @dynamic or provide a method implementation in this category
      [-Wobjc-property-implementation]
@implementation MTLSamplerDescriptor (MoltenVK)
                ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.m:20:
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.h:51:39: note: property declared here
@property(nonatomic, readwrite) float lodBiasMVK;
                                      ^
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.m:33:17: warning: property 'lodBiasMVK' requires method 'setLodBiasMVK:' to be defined - use @dynamic or provide a method implementation in this category
      [-Wobjc-property-implementation]
@implementation MTLSamplerDescriptor (MoltenVK)
                ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.m:20:
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLSamplerDescriptor+MoltenVK.h:51:39: note: property declared here
@property(nonatomic, readwrite) float lodBiasMVK;
                                      ^
2 warnings generated.
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.m:33:17: warning: property 'sampleMaskMVK' requires method 'sampleMaskMVK' to be defined - use @dynamic or provide a method implementation in this
      category [-Wobjc-property-implementation]
@implementation MTLRenderPipelineDescriptor (MoltenVK)
                ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.m:20:
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.h:43:44: note: property declared here
@property(nonatomic, readwrite) NSUInteger sampleMaskMVK;
                                           ^
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.m:33:17: warning: property 'sampleMaskMVK' requires method 'setSampleMaskMVK:' to be defined - use @dynamic or provide a method implementation in this
      category [-Wobjc-property-implementation]
@implementation MTLRenderPipelineDescriptor (MoltenVK)
                ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.m:20:
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/OS/MTLRenderPipelineDescriptor+MoltenVK.h:43:44: note: property declared here
@property(nonatomic, readwrite) NSUInteger sampleMaskMVK;
                                           ^
2 warnings generated.
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:622:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string_view:1059:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/algorithm:1803:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/nth_element.h:15:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:289:52: error: invalid operands to binary expression ('const MVKSmallVectorImpl<BindInfo,
      mvk_smallvector_allocator<BindInfo, 64>>::iterator' and 'difference_type' (aka 'long'))
  const _RandomAccessIterator __leftmost = __first - difference_type(1); (void)__leftmost; // can be unused when assertions are disabled
                                           ~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:759:14: note: in instantiation of function template specialization
      'std::__insertion_sort_unguarded<std::_ClassicAlgPolicy, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58) &, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator>' requested here
        std::__insertion_sort_unguarded<_AlgPolicy, _Compare>(__first, __last, __comp);
             ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:878:8: note: in instantiation of function template specialization
      'std::__introsort<std::_ClassicAlgPolicy, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58) &, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, false>' requested here
  std::__introsort<_AlgPolicy,
       ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:946:10: note: in instantiation of function template specialization
      'std::__sort_dispatch<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
    std::__sort_dispatch<_AlgPolicy>(std::__unwrap_iter(__first), std::__unwrap_iter(__last), __comp);
         ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:954:8: note: in instantiation of function template specialization
      'std::__sort_impl<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
  std::__sort_impl<_ClassicAlgPolicy>(std::move(__first), std::move(__last), __comp);
       ^
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:7: note: in instantiation of function template specialization 'std::sort<MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator,
      (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
        std::sort(sortedBindings.begin(), sortedBindings.end(), [](BindInfo bindInfo1, BindInfo bindInfo2) {
             ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:577:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__ios/fpos.h:61:11: note: candidate template ignored: could not match 'const fpos<_StateT>' against 'const
      MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator'
streamoff operator-(const fpos<_StateT>& __x, const fpos<_StateT>& __y) {
          ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:580:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__iterator/reverse_iterator.h:298:1: note: candidate template ignored: could not match
      'const reverse_iterator<_Iter1>' against 'const MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator'
operator-(const reverse_iterator<_Iter1>& __x, const reverse_iterator<_Iter2>& __y)
^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:581:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__iterator/wrap_iter.h:194:6: note: candidate template ignored: could not match 'const __wrap_iter<_Iter1>'
      against 'const MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator'
auto operator-(const __wrap_iter<_Iter1>& __x, const __wrap_iter<_Iter2>& __y) _NOEXCEPT
     ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:24:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Layers/MVKLayers.h:23:
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKSmallVector.h:103:14: note: candidate function not viable: 'this' argument has type 'const MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator', but
      method is not marked const
    iterator operator- (const diff_type n)   { return iterator( index - n, *vector ); }
             ^
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKSmallVector.h:106:15: note: candidate function not viable: 'this' argument has type 'const MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator', but
      method is not marked const
    diff_type operator- (const iterator& it) { return index - it.index; }
              ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:622:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string_view:1059:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/algorithm:1803:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/nth_element.h:15:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:290:44: error: invalid operands to binary expression ('const MVKSmallVectorImpl<BindInfo,
      mvk_smallvector_allocator<BindInfo, 64>>::iterator' and 'difference_type' (aka 'long'))
  for (_RandomAccessIterator __i = __first + difference_type(1); __i != __last; ++__i) {
                                   ~~~~~~~ ^ ~~~~~~~~~~~~~~~~~~
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:580:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__iterator/reverse_iterator.h:316:1: note: candidate template ignored: could not match
      'reverse_iterator<_Iter>' against 'difference_type' (aka 'long')
operator+(typename reverse_iterator<_Iter>::difference_type __n, const reverse_iterator<_Iter>& __x)
^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:581:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__iterator/wrap_iter.h:206:21: note: candidate template ignored: could not match '__wrap_iter<_Iter1>' against
      'difference_type' (aka 'long')
__wrap_iter<_Iter1> operator+(typename __wrap_iter<_Iter1>::difference_type __n, __wrap_iter<_Iter1> __x) _NOEXCEPT
                    ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:24:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Layers/MVKLayers.h:23:
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKSmallVector.h:101:14: note: candidate function not viable: 'this' argument has type 'const MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator', but
      method is not marked const
    iterator operator+ (const diff_type n)   { return iterator( index + n, *vector ); }
             ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:622:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string_view:1059:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/algorithm:1803:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/nth_element.h:15:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:577:5: error: indirection requires pointer operand ('const MVKSmallVectorImpl<BindInfo,
      mvk_smallvector_allocator<BindInfo, 64>>::iterator' invalid)
    *__begin = _Ops::__iter_move(__pivot_pos);
    ^~~~~~~~
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:802:20: note: in instantiation of function template specialization
      'std::__bitset_partition<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58) &>' requested here
            ? std::__bitset_partition<_AlgPolicy, _RandomAccessIterator, _Compare>(__first, __last, __comp)
                   ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:878:8: note: in instantiation of function template specialization
      'std::__introsort<std::_ClassicAlgPolicy, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58) &, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, false>' requested here
  std::__introsort<_AlgPolicy,
       ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:946:10: note: in instantiation of function template specialization
      'std::__sort_dispatch<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
    std::__sort_dispatch<_AlgPolicy>(std::__unwrap_iter(__first), std::__unwrap_iter(__last), __comp);
         ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:954:8: note: in instantiation of function template specialization
      'std::__sort_impl<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
  std::__sort_impl<_ClassicAlgPolicy>(std::move(__first), std::move(__last), __comp);
       ^
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:7: note: in instantiation of function template specialization 'std::sort<MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator,
      (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
        std::sort(sortedBindings.begin(), sortedBindings.end(), [](BindInfo bindInfo1, BindInfo bindInfo2) {
             ^
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:19:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptor.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKImage.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKResource.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDevice.h:21:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKFoundation.h:23:
In file included from /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/Utility/MVKEnvironment.h:27:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string:622:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/string_view:1059:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/algorithm:1803:
In file included from /Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/nth_element.h:15:
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:647:5: error: indirection requires pointer operand ('const MVKSmallVectorImpl<BindInfo,
      mvk_smallvector_allocator<BindInfo, 64>>::iterator' invalid)
    *__begin = _Ops::__iter_move(__pivot_pos);
    ^~~~~~~~
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:803:20: note: in instantiation of function template specialization
      'std::__partition_with_equals_on_right<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58) &>' requested
      here
            : std::__partition_with_equals_on_right<_AlgPolicy, _RandomAccessIterator, _Compare>(__first, __last, __comp);
                   ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:878:8: note: in instantiation of function template specialization
      'std::__introsort<std::_ClassicAlgPolicy, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58) &, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, false>' requested here
  std::__introsort<_AlgPolicy,
       ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:946:10: note: in instantiation of function template specialization
      'std::__sort_dispatch<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
    std::__sort_dispatch<_AlgPolicy>(std::__unwrap_iter(__first), std::__unwrap_iter(__last), __comp);
         ^
/Applications/Xcode15.3b1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__algorithm/sort.h:954:8: note: in instantiation of function template specialization
      'std::__sort_impl<std::_ClassicAlgPolicy, MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator, (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
  std::__sort_impl<_ClassicAlgPolicy>(std::move(__first), std::move(__last), __comp);
       ^
/Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:7: note: in instantiation of function template specialization 'std::sort<MVKSmallVectorImpl<BindInfo, mvk_smallvector_allocator<BindInfo, 64>>::iterator,
      (lambda at /Users/pip/dev/MoltenVK/MoltenVK/MoltenVK/GPUObjects/MVKDescriptorSet.mm:231:58)>' requested here
        std::sort(sortedBindings.begin(), sortedBindings.end(), [](BindInfo bindInfo1, BindInfo bindInfo2) {
             ^
4 errors generated.
** BUILD FAILED **

make: *** [macos] Error 65
```

--- a/MoltenVK/MoltenVK/Utility/MVKSmallVector.h
+++ b/MoltenVK/MoltenVK/Utility/MVKSmallVector.h
@@ -86,9 +86,9 @@ public:
       return *this;
     }
 
-    Type *operator->() { return &vector->alc.ptr[index]; }
-    Type &operator*()  { return  vector->alc.ptr[index]; }
-    operator Type*()   { return &vector->alc.ptr[index]; }
+    Type *operator->() const { return &vector->alc.ptr[index]; }
+    Type &operator*()  const { return  vector->alc.ptr[index]; }
+    operator Type*()   const { return &vector->alc.ptr[index]; }
 
     bool operator==( const iterator &it ) const { return vector == it.vector && index == it.index; }
     bool operator!=( const iterator &it ) const { return vector != it.vector || index != it.index; }
@@ -98,20 +98,19 @@ public:
     iterator& operator--()      {                 --index; return *this; }
     iterator  operator--( int ) { auto t = *this; --index; return t; }
 
-    iterator operator+ (const diff_type n)   { return iterator( index + n, *vector ); }
-    iterator& operator+= (const diff_type n) { index += n; return *this; }
-    iterator operator- (const diff_type n)   { return iterator( index - n, *vector ); }
-    iterator& operator-= (const diff_type n) { index -= n; return *this; }
+    iterator operator+ (const diff_type n)   const { return iterator( index + n, *vector ); }
+    iterator& operator+= (const diff_type n)       { index += n; return *this; }
+    iterator operator- (const diff_type n)   const { return iterator( index - n, *vector ); }
+    iterator& operator-= (const diff_type n)       { index -= n; return *this; }
 
-    diff_type operator- (const iterator& it) { return index - it.index; }
+    diff_type operator- (const iterator& it) const { return index - it.index; }
 
-    bool operator< (const iterator& it)  { return index < it.index; }
-    bool operator<= (const iterator& it) { return index <= it.index; }
-    bool operator> (const iterator& it)  { return index > it.index; }
-    bool operator>= (const iterator& it) { return index >= it.index; }
+    bool operator< (const iterator& it)  const { return index < it.index; }
+    bool operator<= (const iterator& it) const { return index <= it.index; }
+    bool operator> (const iterator& it)  const { return index > it.index; }
+    bool operator>= (const iterator& it) const { return index >= it.index; }
 
-    const Type &operator[]( const diff_type i ) const { return vector->alc.ptr[index + i]; }
-    Type &operator[]( const diff_type i )             { return vector->alc.ptr[index + i]; }
+    Type &operator[]( const diff_type i ) const { return vector->alc.ptr[index + i]; }
 
     bool   is_valid()     const { return index < vector->alc.size(); }
     size_t get_position() const { return index; }

